### PR TITLE
Prepare for 1.6.1 release

### PR DESCRIPTION
**Pull request by rouault**
@thbeu If you want to have a look at the changelog to check it reflects correctly the changes

### CODE REVIEW

File: web/release.html [hunk 1]
```diff
@@ -10,6 +10,39 @@ <h1>Shapelib Release Notes</h1>
...
+<li>DBFIsValueNULL(): Accept string containing of width times "0" as NULL Date</li>
+<li>Only test for _WIN32 for Windows detection</li>
+<li>Windows: Fix UTF8 hook functions</li>
+<li>Various compiler warning fixes</li>
+<li>contrib/csv2shp.c: fix resource leaks (#86)</li>
+<li>Detect byte order at compile time</li>
+<li>shapefil.h: various const-correctness improvements</li>
+<li>shapefil.h: Remove SHPTreeRemoveShapeId not being implemented</li>
+<li>shapefil.h: Fix API call of SHPSearchDiskTreeEx</li>
+<li>Add wrapper functions for read/write of date attributes</li>
```
Comment (thbeu) [inline web/release.html:25]:
```suggestion
<li>Add API functions for read/write of date attributes: DBFReadDateAttribute/DBFWriteDateAttribute</li>
```

**Review by thbeu — CHANGES_REQUESTED** [review]
I miss the two "breaking" changes:

* 94450f9a04699bd74d7d88e65bb1475266b59f73: DBFWriteAttribute/DBFWriteLogicalAttribute no longer silently accpets invalid input, bur returns false
* 5b8c378f98b25b6c880496587f8f052a661b1f7a: DBFCloneEmpty considers the SAHooks

File: web/release.html [hunk 2]
```diff
@@ -10,6 +10,39 @@ <h1>Shapelib Release Notes</h1>
...
 the project mailing list at https://lists.osgeo.org/pipermail/shapelib/.<p>
 
+<b>Release 1.6.1</b>:
+<ul>
+<li>DBFIsValueNULL(): accept empty string as NULL Date (https://github.com/OSGeo/gdal/issues/10405)</li>
+<li>DBFIsValueNULL(): Accept string containing of width times "0" as NULL Date</li>
+<li>Only test for _WIN32 for Windows detection</li>
+<li>Windows: Fix UTF8 hook functions</li>
+<li>Various compiler warning fixes</li>
+<li>contrib/csv2shp.c: fix resource leaks (#86)</li>
```
[the remark this answers is not in the document]
Comment (rouault) [reply web/release.html:20]: I use what is available in commit messages.

File: web/release.html (hunk 1 above)
Comment (rouault) [reply web/release.html]: done

**Comment by rouault** [comment]
> I miss the two "breaking" changes:

thanks, added (the messages you propose here make much more sense to me than the raw commit messages I use to build the change log, and I tend to skip the ones I can't immediately make sense of, so this is an encouragement in providing more descriptive commit messages :-))

File: web/release.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (thbeu) [reply web/release.html:20]: Do you know the [auto-release notes generation feature of GitHub](https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes)? It helps to keep track of the PR based changes.

**Pull request merged**